zdm/grid.py

```diff
@@ -628,26 +656,41 @@ def update(self, vparams:dict, ALL=False):
         # TODO -- For cubes with multiple surveys can we do these
         #   first two steps (even the first 5!) only once??
         # Update cosmology?
-        if reset_cos:
+        if reset_cos and prev_grid is None:
             cos.set_cosmology(self.state)
             cos.init_dist_measures()
 
         if get_zdm or ALL:
-            zDMgrid, zvals,dmvals=misc_functions.get_zdm_grid(
-                self.state, new=True,plot=False,method='analytic')
-            # TODO -- Check zvals and dmvals haven't changed!
-            self.pass_grid(zDMgrid,zvals,dmvals)
+            if prev_grid is None:
+                zDMgrid, zvals,dmvals=misc_functions.get_zdm_grid(
+                    self.state, new=True,plot=False,method='analytic')
+                self.pass_grid(zDMgrid,zvals,dmvals)
+            else:
+                # Pass a copy (just to be safe)
```

Comment:
Agreed that copying is safer, I came unstuck by this at one point during development and it was a pain.